packages/plugin-credentials/src/token.js

```diff
@@ -0,0 +1,332 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+/* eslint camelcase: [0] */
+
+import {pick} from 'lodash';
+import {oneFlight, tap} from '@ciscospark/common';
+import {grantErrors, SparkPlugin} from '@ciscospark/spark-core';
+import {sortScope} from './scope';
+
+/**
+ * Parse response from CI and converts to structured error when appropriate
+ * @param {SparkHttpError} res
+ * @private
+ * @returns {GrantError}
+ */
+function processGrantError(res) {
+  if (res.statusCode !== 400) {
+    return Promise.reject(res);
+  }
+
+  const ErrorConstructor = grantErrors.select(res.body.error);
+  return Promise.reject(new ErrorConstructor(res._res || res));
+}
+
+const Token = SparkPlugin.extend({
+  namespace: `Credentials`,
+
+  props: {
```

Comment:
not really. a Token is usable if it as as little as just a refresh token or just an access token